CRM/Core/BAO/FinancialTrxn.php

```diff
@@ -696,4 +696,30 @@ public static function createDeferredTrxn($lineItems, $contributionDetails, $upd
     }
   }
 
+  /**
+   * Format Credit Card type.
+   *
+   * @param array $params
+   *
+   */
+  public static function formatCreditCardDetails(&$params) {
+    $creditCardType = CRM_Utils_Array::value('credit_card_type', $params);
+    if ($creditCardType) {
+      if (!is_numeric($creditCardType)) {
+        $creditCardTypes = CRM_Core_PseudoConstant::get('CRM_Financial_DAO_FinancialTrxn',
```

Comment:
I used CRM_Core_PseudoConstant::getKey() which checks against machine name and retrieves value of option values.